simulation/amaru-sim/src/simulator/simulate.rs

```diff
@@ -262,41 +286,29 @@ pub fn simulate<Msg, F>(
 {
     let mut rng = StdRng::seed_from_u64(config.seed);
 
-    for test_number in 0..config.number_of_tests {
+    for test_number in 1..=config.number_of_tests {
         let entries: Vec<Reverse<Entry<Msg>>> = generator(&mut rng);
 
-        let node_handles: Vec<_> = (1..=config.number_of_nodes)
-            .map(|i| (format!("n{}", i), spawn()))
-            .collect();
-
-        let mut world = World::new(entries.clone(), node_handles);
-
-        match world.run_world() {
-            Err((reason, history)) => {
+        match run_test(config.number_of_nodes, &spawn, &property)(&entries) {
```

Comment:
can you reuse `run_test(..)` here, eg. something like 
```
let test = run_test(config.number_of_nodes, &spawn, &property)
match test(&entries) {
...
}
```

Did not try it so take with a grain of salt